cli/command/system/prune_progress.go

```diff
@@ -0,0 +1,173 @@
+package system
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/progress"
+	"github.com/docker/docker/pkg/streamformatter"
+)
+
+type pruneProgressChan struct {
+	isEventsListening chan bool
+	errorChan         chan error
+}
+
+func runPruneProgress(dockerCli command.Cli, progressChans *pruneProgressChan, options pruneOptions) {
+	eventChan := make(chan string, 1)
+	objCount, err := countObjectsToRemove(dockerCli, options)
+	if err != nil {
+		progressChans.errorChan <- err
+		return
+	}
+	go monitorRemovingEvents(dockerCli, eventChan, progressChans)
```

Comment:
I think monitoring needs to start before counting the objects. Otherwise anything that is created or removed after the count will mess with the progress.

Reply:
Well, we have a synchronization bit called `isEventsListening` that is used to authorize the pruning process when the monitoring is up and running.